### ISSUE TITLE
Reduce `max_attempts` default value

### ISSUE THREAD

**Issue by shhlife** [griptape-ai/griptape#1439]
- [x] I have read and agree to the [contributing guidelines](https://github.com/griptape-ai/griptape#contributing).

Currently `max_attempts` is set to 10, which can make it appear that the framework is hung when it's really just trying something a number of times.

Would like to discuss reducing the default number to 2 or 3. Usually if something is going to fail, one or two retries is fine - and if the user needs more they can always increase it.

**Comment by collindutter** [comment]
I wonder if we should retry at all? Should it be opt-in?

**Comment by shhlife** [comment]
good point, maybe not? Usually things fail due to something that retrying wouldn't resolve.. like an invalid api key.

**Comment by collindutter** [comment]
Or maybe we just need to refine the [ignored exception types](https://github.com/griptape-ai/griptape/blob/36d5d2c0c22ffabc85b7e017952582905ee94554/griptape/drivers/prompt/openai_chat_prompt_driver.py?plain=1#L84-L96).
